victor/src/layout/flow/mod.rs

```diff
@@ -139,19 +138,24 @@ impl BlockLevelBox {
         absolutely_positioned_fragments: &mut Vec<AbsolutelyPositionedFragment<'a>>,
     ) -> Fragment {
         match self {
-            BlockLevelBox::SameFormattingContextBlock { style, contents } => {
-                same_formatting_context_block(
+            BlockLevelBox::SameFormattingContextBlock { style, contents } => in_flow_non_replaced(
+                containing_block,
+                absolutely_positioned_fragments,
+                style,
+                move |containing_block| contents.layout(containing_block, tree_rank),
+            ),
+            BlockLevelBox::Independent { style, contents } => match contents.as_replaced() {
+                Ok(replaced) => {
+                    // FIXME
+                    match *replaced {}
+                }
+                Err(contents) => in_flow_non_replaced(
                     containing_block,
-                    tree_rank,
                     absolutely_positioned_fragments,
                     style,
-                    contents,
-                )
-            }
-            BlockLevelBox::Independent { style: _, contents } => {
-                // FIXME
-                match *contents {}
-            }
+                    move |containing_block| contents.layout(containing_block),
```

Comment:
I am really confused about this code, we don't pass `tree_rank` so what happens to abspos elements in independent formatting contexts?

Reply:
`tree_rank` is used in `BlockContainer::layout` to mark abspos elements in this block container apart from those in siblings. `BlockFormattingContext::layout` does not take a `tree_rank` parameter and instead uses `let dummy_tree_rank = 0;`. When a block formatting context contains a block container it contains nothing else, so there is no sibling to distinguish from.

Reply:
So let's say there is an IndependentFormattingContext somewhere in a BlockContainer, how is the tree rank of that independent formatting context propagated to its abspos descendants?

Reply:
I feel like somewhere, we are lacking some tree rank propagation for [this code](https://github.com/SimonSapin/victor/blob/03888c56613bc305e8d40c571a08ed9d0a347d31/victor/src/layout/flow/mod.rs#L81-L87) to work properly.

Reply:
We can discuss about that tomorrow.

Reply:
I think you’re right, we’re missing code is some places both to count the rank and to adjust static positions to account for coordinate origin change.

Reply:
I have a commit for it, which is nice.